modules/ctl/src/main/scala/command.scala

```diff
@@ -37,116 +35,119 @@ object Command {
    * name of the executable ("gemctl" for example).
    */
   def parse[A](progName: String, args: List[String]): IO[Option[Command]] =
-    execParserPure(prefs(idm[PrefsMod]), mainParser, args) match {
-      case Success(c) => IO(Some(c))
-      case Failure(f) =>
-        import Predef._
-        val (msg, exit) = renderFailure(f, progName)
-        IO.putStr(Console.BLUE) *>
-        IO.putStrLn(msg) *>
-        IO.putStrLn("")  *>
-        IO.putStrLn(s"""
-          |Hints:
-          |  $progName --help            To see available commands.
-          |  $progName COMMAND --help    To see help on a specific command.
-         """.trim.stripMargin) *>
-         IO.putStr(Console.RESET) as None
+    mainParser(progName).parse(args) match {
+      case Right(c) => c.some.pure[IO]
+      case Left(h) => IO(Console.println(Console.BLUE + h.toString + Console.RESET)).as(none[Command]) // scalastyle:ignore
     }
 
-  // Parser implementation below
-
-  private lazy val machine: Parser[Boolean] =
-    switch(
-      short('m'), long("machine"),
-      help("Use docker machine. Use with -H to specify a machine other than 'default'.")
-    )
-
-  private lazy val host: Parser[Option[String]] =
-    optional(
-      strOption(
-        short('H'), long("host"), metavar("HOST"),
-        help("Use the specified docker host (machine when given with -m).")
-      )
-    )
-
-  private lazy val server: Parser[Server] =
-    (machine |@| host |@| user) {
+  // Opts implementation below
+
+  private lazy val machine: Opts[Boolean] =
+    Opts.flag(
+      short = "m",
+      long  = "machine",
+      help  = "Use docker machine. Use with -H to specify a machine other than 'default'."
+    ).orFalse
+
+  private lazy val host: Opts[Option[String]] =
+    Opts.option[String](
+      short   = "H",
+      long    = "host",
+      metavar = "HOST",
+      help    = "Use the specified docker host (machine when given with -m)."
+    ).orNone
+
+  private lazy val server: Opts[Server] =
+    (machine |@| host |@| user) map {
```

Comment:
Maybe not the place to ask because here the applicative syntax fits nicely, but now that I see this, do you know if there is anything to make for-comprehesions desugar to applicatives when possible?